### DISCUSSION TITLE
Eventual Atomixwap implementation in ASA Stats

### DISCUSSION THREAD

**Discussion by ipaleka** [asastats/channel#139] [General]
https://discord.com/channels/906917846754418770/973245314687434832/987442715488837723

- - - - -

Atomixwap is now live on mainnet
https://twitter.com/Atomixwap/status/1536390953472335875

https://twitter.com/Atomixwap
https://atomixwap.xyz/

[Atomixwap (@Atomixwap)](https://twitter.com/Atomixwap)
🟢 Atomixwap is now live on mainnet 🟢
A new alternative to generate you swap links!

- MyAlgo & Pera support
- Accept payment in $DEGEN (more to come)
- Variable royalty for creator (0-10%)
- Free to use
- No smart contract involved

![FVJavF6XoAIDqFZ](https://user-images.githubusercontent.com/49662536/174392745-def55bf3-e77e-4655-97e8-f6735faa509c.jpg)

Twitter•06/13/2022
[Atomixwap (@Atomixwap)](https://twitter.com/Atomixwap)
Trade NFTs for Algo or other predefined cryptocurrency on the Algorand Blockchain using atomic tranfers. Coded by @LionsWrinkle.
https://t.co/GI1qixZ7Ix

- - - - -

Do we need to implement such a project in ASA Stats and which information we can provide to our users?

At first, the receiver's address can get a "reserved" entry for the NFT in a group transaction.

**Comment by wxivd** [comment]
> Do we need to implement such a project in ASA Stats?

I would really support that.

- [RandSwap](https://twitter.com/Atomixwap/status/1536444222605238284?s=20&t=kTZ5OnhPC1f8-AytznyU3w) is now redirecting to Atomixwap
- The tool is free to use
- The tool serves an important purpose: sending/Trading NFTs

> which information we can provide to our users?
- Text: Trade NFT using Atomixwap. 
- A hyperlink to the above text. 
-  Display under the Last/Max purchase / Floor price section.

**Reply by ipaleka** [reply]
> > Do we need to implement such a project in ASA Stats?
> 
> I would really support that.
> 
> * [RandSwap](https://twitter.com/Atomixwap/status/1536444222605238284?s=20&t=kTZ5OnhPC1f8-AytznyU3w) is now redirecting to Atomixwap
> 
> * The tool is free to use
> 
> * The tool serves an important purpose: sending/Trading NFTs

Ok, fair enough. 

> > which information we can provide to our users?
> 
> * Text: Trade NFT using Atomixwap.
>
> * A hyperlink to the above text.
> 
> * Display under the Last/Max purchase / Floor price section.

ASA Stats isn't an advertising platform. Afaic if someone has created a group transaction (an atomic swap) for an NFT that address holds, then for those [75 minutes](https://twitter.com/Atomixwap/status/1537360415872667650) after creation we can create a link to that atomic swap on Atomixwap.

**Reply by wxivd** [reply]
> ASA Stats isn't an advertising platform.

I did not see it as an advertisement. Hence it is free. I can see your point though. 

> if someone has created a group transaction (an atomic swap) for an NFT that address holds, then for those [75 minutes](https://twitter.com/Atomixwap/status/1537360415872667650) after creation we can create a link to that atomic swap on Atomixwap.

That could potentially become a form of advertisement? If I like a given NFT (I do not own), then I can create a link, propose a value. If ASA Stats display the link to NFT owner then that would be a sort of "offer" received.

**Reply by ipaleka** [reply]
> ASA Stats isn't an advertising platform.

I should elaborate on this. If we add Atomixwap link then we have to add all the NFT markets' links too. And that implies that we have to add all the AMMs and dexes links for all the *regular* ASAs too.

Those extra links are directly against the "best UX" goal, as well as the advertising itself would be. Nevertheless, that doesn't mean ASA Stats won't go that way one day, but I as a member of the Team can't guarantee to finish the roadmap before the deadline if we step on that path before DAO takes over as that brings a whole new level of tasks and obligations.

**Reply by ipaleka** [reply]
> > ASA Stats isn't an advertising platform.
> 
> I did not see it as an advertisement. Hence it is free. I can see your point though.

This can become a philosophical debate I want to avoid. :) In short, there's no such thing as "free" in this (software) industry.

> > if someone has created a group transaction (an atomic swap) for an NFT that address holds, then for those [75 minutes](https://twitter.com/Atomixwap/status/1537360415872667650) after creation we can create a link to that atomic swap on Atomixwap.
> 
> That could potentially become a form of advertisement? If I like a given NFT (I do not own), then I can create a link, propose a value. If ASA Stats display the link to NFT owner then that would be a sort of "offer" received.

Absolutely. ASA Stats goal is to present all information about every ASA and NFT on the Algorand blockchain. The only thing we don't do is creating an information by ourselves.

**Comment by ipaleka** [comment]
Whatever we decide to do with the Atomixwap links, we still have to implement those transactions into our engine as those transactions represent sales just like any other NFT market.

https://algoexplorer.io/tx/group/BU4P0Cxyh5ThaZrk343Z58qGJNTtQCgmWSEmKI9UmmQ%3D

**Comment by ipaleka** [comment]
> Afaic if someone has created a group transaction (an atomic swap) for an NFT that address holds, then for those [75 minutes](https://twitter.com/Atomixwap/status/1537360415872667650) after creation we can create a link to that atomic swap on Atomixwap.

I'm afraid this functionality isn't reachable to us by design. It seems that there are no database records at all as all the data needed is inside the link itself. That means the provider can't provide any API-like mechanism for us. 

Basically, even the creator can't get access to the future atomic swap if they lose the link, they have to create it again.

Still, after the targeted party signs a group transaction, it is, of course, available onchain and we have to [implement Atomixwap](https://github.com/asastats/channel/issues/141) as an NFT market having only purchases, without current listings.

**Comment by ipaleka** [comment]
This kind of application was the reason why we [started to think](https://discord.com/channels/906917846754418770/914899975182364672/939095953527410759) about standardization:

- - - - -

So, first a little introduction, ARC3 (https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0003.md) is a standard defined by Algorand (by Fabrice to be exact) to use for NFT creation and because of some of the problems in Algorand community those guys created their own arc69 standard for NFT on Algorand. Meanwhile, a common standard has emerged too (https://github.com/algorandfoundation/ARCs/pull/38#issuecomment-1028424527).

So, that's for NFTs, but we need something different. Let me first explain why.

There are some NFTs in the Algorand space that have not ever been sold. If they are created by some established creator then we can extrapolate the approximated value of such NFT from the values of other NFT in the collection. But for example if the all collection has been gifted, we can't source price information as we don't have aby basis.

What we can do in theory? We can track swaps. The easiest example to understand is if some NFT was swapped for 100 ALGO then we can easily set the value to 100 ALGO.

The more complex example is if 2 NFTs are swapped for 100 ALGO. We then have sum value and if we have a value for one of them from NFT market then we can set the value for the other too.

And the really complex example is if 5 NFT are swapped for 2 other NFTs and not even one of them has ever been sold on any NFT market. We then track each of the involved NFTs further in the history of atomic swaps until we get all the missing variables.

So, the main problem is that we don't have information that tells us "Look, this group transaction is created by well known and established Algorand swap provider website/software".

So we need to create the initiative so that swap providers start to mark their atomic swaps so we can track them in real-time.

- - - - -

For example, Swapper (https://app.swapper.tools) leaves no trace at all, so we can't collect anything.

Atomixwap at least leaves a note so we can check if the group indeed comes from them.

**Reply by wxivd** [reply]
Playing the devil advocate here a malpractice that can potentially inflate the estimated value by simply swapping two NFTs between wallets owned by same user. Especially in tools that are free to use which removes a swapping fee barrier despite it being small

**Reply by ipaleka** [reply]
> Playing the devil advocate here a malpractice that can potentially inflate the estimated value by simply swapping two NFTs between wallets owned by same user. Especially in tools that are free to use which removes a swapping fee barrier despite it being small

It is implemented (will be deployed soon). Zero value shouldn't affect future price evaluation.

**Comment by ipaleka** [comment]
https://twitter.com/asastatscom/status/1541771293313441795

> We have implemented Atomixwap as an NFT market provider. We're using this opportunity to call for the creation of a standard that will be used by swap tools providers to identify NFT and other ASA swap transactions: 
https://github.com/algorandfoundation/ARCs/issues/110 [#asastats](https://twitter.com/hashtag/asastats?src=hashtag_click) [@Atomixwap](https://twitter.com/Atomixwap) [@Silventino](https://twitter.com/Silventino)
